packages/react/src/components/Table/index.js

```diff
@@ -0,0 +1,158 @@
+    })}
+    {...props}
+  >
+    {children}
+  </th>
+);
+
+HeaderCell.propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.node,
+  isSorted: PropTypes.bool,
+};
+
+HeaderCell.defaultProps = {
+  className: undefined,
+  children: undefined,
+  isSorted: false,
+};
+
+const RowGroup = ({ className, children, ...props }) => (
+  <tr className={cx("Table-RowGroup", className)} {...props}>
+    {children}
+  </tr>
+);
+
+RowGroup.propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.node,
+};
+
+RowGroup.defaultProps = {
+  className: undefined,
+  children: undefined,
+};
+
+const RowCell = ({ className, children, ...props }) => (
+  <td className={cx("Table-RowCell", className)} {...props}>
+    {children}
+  </td>
+);
+
+RowCell.propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.node,
+};
+
+RowCell.defaultProps = {
+  className: undefined,
+  children: undefined,
+};
+
+const Table = ({ className, columns, data, isSortable }) => {
+  const { headerGroups, rows, prepareRow } = useTable(
+    {
+      columns,
+      data,
+    },
+    useSortBy
+  );
+
+  return (
+    <table className={cx("Table", className)}>
+      <thead>
+        {headerGroups.map(({ getHeaderGroupProps, headers }) => (
+          <HeaderGroup {...getHeaderGroupProps()}>
+            {headers.map(
+              ({
+                getHeaderProps,
+                getSortByToggleProps,
+                isSorted,
+                isSortedDesc,
+                render,
+              }) => (
+                <HeaderCell
+                  {...getHeaderProps(isSortable && getSortByToggleProps())}
+                  isSorted={isSorted}
+                >
+                  <Text>
+                    {render("Header")}
+                    {isSorted && isSortedDesc && <Icon icon="IconArrowUp" />}
+                    {isSorted && !isSortedDesc && <Icon icon="IconArrowDown" />}
+                    {isSortable && !isSorted && (
+                      <Icon icon="IconChevronUpDown" />
+                    )}
+                  </Text>
+                </HeaderCell>
+              )
+            )}
+          </HeaderGroup>
+        ))}
+      </thead>
+
+      <tbody>
+        {rows.map(row => (
+          <RowGroup {...prepareRow(row)} {...row.getRowProps()}>
+            {row.cells.map(({ getCellProps, render }) => (
+              <RowCell {...getCellProps()}>{render("Cell")}</RowCell>
+            ))}
+          </RowGroup>
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+Table.propTypes = {
+  className: PropTypes.string,
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      Header: PropTypes.string.isRequired,
+      accessor: PropTypes.string.isRequired,
+    })
+  ),
+  data: PropTypes.arrayOf(PropTypes.shape({})),
+  isSortable: PropTypes.bool,
+};
+
+Table.defaultProps = {
+  className: undefined,
+  columns: [],
+  data: [],
+  isSortable: false,
+};
+
+Table.HeaderGroup = HeaderGroup;
```

Comment:
Ugo, just because this file grew this big, shouldn't each of these components be in a separate file,
and then this file would be doing 3 imports and then exporting everything as from one file?

Reply:
Create 3 files for less 10 lines each, IDK if it's worth it

Reply:
Saving couple of lines is one thing, but readability of the file itself could be of consequence.
As in, the navigation across this file is not as easy as it could be.
@obedparla @keanan what are you thoughts on this?